blockmanager.go

```diff
@@ -425,16 +429,20 @@ func (b *blockManager) handleDonePeerMsg(peers *list.List, p *peer) {
 
 	// Remove requested transactions from the global map so that they will
 	// be fetched from elsewhere next time we get an inv.
-	for k := range p.requestedTxns {
-		delete(b.requestedTxns, k)
+	for k, txPeer := range b.requestedTxns {
```

Comment:
Also, in regards to this, there really needs to be an overall map in addition to the described one.  That way it's trivial to look up by requested hash or peer.  So something like:

```Go
requestedTxns:   make(map[wire.ShaHash]struct{}),
requestedPeerTxns:   make(map[int64]map[wire.ShaHash]struct{}),
```